lib/analyzer/static-analyzer.ts

```diff
@@ -192,12 +201,30 @@ export async function analyze(
     [];
 
   if (appScan) {
+    const collectApplicationFiles = isTrue(
+      options["collect-application-files"],
+    );
+
     const nodeDependenciesScanResults = await nodeFilesToScannedProjects(
       getFileContent(extractedLayers, getNodeAppFileContentAction.actionName),
     );
+    let nodeApplicationFilesScanResults: AppDepsScanResultWithoutTarget[] = [];
+    if (collectApplicationFiles) {
+      nodeApplicationFilesScanResults = getApplicationFiles(
+        getFileContent(
+          extractedLayers,
+          getNodeJsTsAppFileContentAction.actionName,
+        ),
+        "node",
+        "npm",
+        isNodeAppFile,
```

Comment:
No need for the function since the filtering is done in the getFileContent function already